### PR TITLE
[SDPA-2673] Enable the field_media_site field

### PR DESCRIPTION
**Pull request by vincent-gao**
~removing `config/install/field.storage.media.field_media_site.yml` from tide_site module , this config~ ~should be in tide_media.~
the tide_site module suddenly gets an error on content-vic test.
```bash
You are about to DROP all tables in your 'drupal' database. Do you want to continue? (y/n): y
Starting Drupal installation. This takes a while. Consider using the        [ok]
--notify global option.
Drupal\Core\Config\PreExistingConfigException: Configuration objects     [error]
(field.storage.media.field_media_site) provided by tide_site already
exist in active configuration in
/app/docroot/core/lib/Drupal/Core/Config/PreExistingConfigException.php:65
Stack trace:
#0 /app/docroot/core/lib/Drupal/Core/Config/ConfigInstaller.php(496):
Drupal\Core\Config\PreExistingConfigException::create('tide_site',
Array)
#1
/app/docroot/core/lib/Drupal/Core/ProxyClass/Config/ConfigInstaller.php(132):
Drupal\Core\Config\ConfigInstaller->checkConfigurationToInstall('module',
'tide_site')
#2
/app/docroot/core/lib/Drupal/Core/Extension/ModuleInstaller.php(150):
Drupal\Core\ProxyClass\Config\ConfigInstaller->checkConfigurationToInstall('module',
'tide_site')
#3
/app/docroot/core/lib/Drupal/Core/ProxyClass/Extension/ModuleInstaller.php(83):
Drupal\Core\Extension\ModuleInstaller->install(Array, false)
#4 /app/docroot/core/includes/install.core.inc(1894):
Drupal\Core\ProxyClass\Extension\ModuleInstaller->install(Array,
false)
#5 /app/docroot/core/includes/batch.inc(294):
_install_module_batch('tide_site', 'Tide site', Array)
```
More details about failed test. https://circleci.com/gh/dpc-sdp/content-vic-gov-au/9157?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

it doesn't make sense. the [[content-vic] PR](https://github.com/dpc-sdp/content-vic-gov-au/pull/720) didn't touch *tide_site* module, but it still got the error above. for fixing the error I added
```yml
  enforced:
    module:
      - tide_site
```
into **field.storage.media.field_media_site.yml** . see https://github.com/dpc-sdp/tide_site/pull/40/files

relates to https://github.com/dpc-sdp/content-vic-gov-au/pull/720 and https://github.com/dpc-sdp/tide_media/pull/27

### CODE REVIEW

File: tide_site.install [hunk 1]
```diff
@@ -185,3 +185,28 @@ function tide_site_update_8006() {
...
+  $default_option = [
+    'type' => 'options_buttons',
+    'region' => 'content',
+    'settings' => [],
+    'third_party_settings' => [],
+  ];
+  $bundle_info_service = \Drupal::service('entity_type.bundle.info');
+  $entity_form_display_storage = \Drupal::entityTypeManager()->getStorage('entity_form_display');
+  $media_bundles = $bundle_info_service->getBundleInfo('media');
+  if ($media_bundles) {
```
Comment (anthony-malkoun) [inline tide_site.install]: Just a small comment about style, I prefer the condition to be explicit. Is `$media_bundles` a boolean or an array (an array if you look further) so something more explicit is preferable.

**Comment by GROwen** [comment]
@vincent-gao Why was field_media_site removed from this module into tide_media? It makes sense to have this module define the field because of the dependency on site functionality. Two things should be done to do this:

1. add the `field.storage` config file
2. add the field to the optional property of config_devel 

Also if you're going to make changes to a PR that's already been approved you should request re-review from the users who have previously reviewed, or add a comment to describe why the change was made.

File: tide_site.install (hunk 1 above)
Comment (vincent-gao) [reply tide_site.install]: Hi @anthony-malkoun 
```
if (!empty($media_bundles) && is_array($media_bundles)) {
```
will be more explicit, i think

**Comment by vincent-gao** [comment]
> @vincent-gao Why was field_media_site removed from this module into tide_media? It makes sense to have this module define the field because of the dependency on site functionality. Two things should be done to do this:
> 
> 1. add the `field.storage` config file
> 2. add the field to the optional property of config_devel
> 
> Also if you're going to make changes to a PR that's already been approved you should request re-review from the users who have previously reviewed, or add a comment to describe why the change was made.

@GROwen yes. I think I should close down the tide_media PR https://github.com/dpc-sdp/tide_media/pull/27 , which doesn't make sense to have field.storage.media.field_media_site.yml in tide_media.

**Pull request closed without merging**